modules/codemod/lib/v9/compoundToast.ts

```diff
@@ -0,0 +1,213 @@
+          j.jsxIdentifier('icon'),
+          j.jsxExpressionContainer(j.jsxIdentifier('checkIcon'))
+        ),
+      ];
+
+      const unchangedIconAttributes = attributes?.filter(
+        attr =>
+          attr.type === 'JSXAttribute' &&
+          typeof attr.name.name === 'string' &&
+          /icon|iconColor/g.test(attr.name.name)
+      );
+
+      const updatedIconAttrs = unchangedIconAttributes?.map(specifier => {
+        if (specifier.type === 'JSXAttribute' && specifier.name.name === 'iconColor') {
+          specifier.name.name = 'color';
+        }
+        return specifier;
+      });
+
+      const chooseIconAttrs =
+        unchangedIconAttributes && unchangedIconAttributes.length
+          ? updatedIconAttrs
+          : defaultIconAttributes;
+
+      const closeAttributes = attributes?.filter(
+        attr => attr.type === 'JSXAttribute' && attr.name.name === 'onClose'
+      );
+
+      const linkAttributes = attributes?.filter(
+        attr =>
+          attr.type === 'JSXAttribute' &&
+          (attr.name.name === 'actionText' || attr.name.name === 'onActionClick')
+      );
+
+      // Elements
+
+      const OpeningIconJSX = j.jsxOpeningElement(ToastIconJSX, chooseIconAttrs, true);
+
+      const IconElement = j.jsxElement(OpeningIconJSX);
+
+      // Add `import {checkIcon} from '@workday/canvas-system-icons-web'`
+      // when there's a default toast
+      const lastImport = root.find(j.ImportDeclaration).at(-1);
+      if (lastImport && !unchangedIconAttributes?.length) {
+        lastImport.insertAfter(
+          j.importDeclaration(
+            [j.importSpecifier(j.identifier('checkIcon'))],
+            j.stringLiteral('@workday/canvas-system-icons-web')
+          )
+        );
+      }
+
+      const lineBreakStart = j.jsxText('\n  ');
+      const lineBreakEnd = j.jsxText('\n');
+
+      // Filter out all old props
+      const filteredAttrNames = [
+        'actionText',
+        'icon',
+        'iconColor',
+        'onActionClick',
+        'onClose',
+        'color',
+      ];
+
+      const filterAttr = nodePath.value.openingElement.attributes?.filter(attr => {
+        return !(
+          attr.type === 'JSXAttribute' &&
+          typeof attr.name.name === 'string' &&
+          filteredAttrNames.includes(attr.name.name)
+        );
+      });
+
+      nodePath.value.openingElement.attributes = filterAttr;
+
+      const OpeningCloseElement = j.jsxOpeningElement(ToastCloseJSX, closeAttributes, true);
+
+      const CloseElement = j.jsxElement(OpeningCloseElement);
+
+      // grab the onActionClick attr
+      const onActionClickAttr = linkAttributes?.filter(
+        attr => attr.type === 'JSXAttribute' && attr.name.name === 'onActionClick'
+      );
+
+      // rename onActionClick to onClick for the Toast.Link component
+      const renameOnActionClick = onActionClickAttr?.map(attr => {
+        if (attr.type === 'JSXAttribute' && attr.name.name === 'onActionClick') {
```

Comment:
I've just noticed that you don't need to have `&& attr.name.name === 'onActionClick'` in `.map` because you already filtered all items by this name (line 143).